.github/workflows/check-docs-pr-preview.yml

```diff
@@ -0,0 +1,47 @@
+# This workflow checks that PRs modifying website docs include a preview link in the PR description.
+# It enforces KubeStellar's documentation PR best practices.
+
+name: Check Docs PR Preview Link
+
+on:
+  pull_request:
+    types: [opened, edited, synchronize]
+    paths:
+      - 'docs/content/**'
+      - 'docs/README.md'
+      - 'docs/mkdocs.yml'
+      - 'docs/overrides/**'
+      - 'docs/scripts/**'
+      - 'docs/requirements.txt'
+      - 'docs/main.py'
+
+jobs:
+  check-preview-link:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+      - name: Check for Preview Link in PR Description (bash)
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PR_NUMBER: ${{ github.event.pull_request.number }}
+          REPO: ${{ github.repository }}
+          BRANCH: ${{ github.head_ref }}
+        run: |
+          set -e
+          BODY=$(gh pr view "$PR_NUMBER" --repo "$REPO" --json body -q '.body')
+          DOC_BRANCH_REGEX='^doc-'
+          GITHUB_PAGES_URL="https://$(echo $REPO | cut -d'/' -f1).github.io/$(echo $REPO | cut -d'/' -f2)/$BRANCH"
+          PREVIEW_LINE=$(echo "$BODY" | grep -iE '^\s*Preview.*https?://')
+          if [ -n "$PREVIEW_LINE" ]; then
+            if [[ "$BRANCH" =~ $DOC_BRANCH_REGEX ]] && [[ "$PREVIEW_LINE" == *"$GITHUB_PAGES_URL"* ]]; then
+              echo "\u2705 Docs PR: doc- branch and valid preview link present."
+            elif [[ ! "$BRANCH" =~ $DOC_BRANCH_REGEX ]]; then
+              echo "::warning::\u26a0\ufe0f Docs PR: Preview link found, but branch does not start with doc-. Please consider using a doc- branch for automatic previews."
+            else
+              echo "::warning::\u26a0\ufe0f Docs PR: Preview link found, but it may not point to the expected GitHub Pages URL."
+            fi
+          else
+            echo -e "::error::\u274c Docs PR: No preview link found. Please include a line starting with 'Preview' and a valid GitHub Pages URL for your changes.\n"
```

Comment:
My complaint about the terminal linebreak is for the whole file, not this error message. See the red circle-dash at the end of GitHub's display here?
<img width="248" alt="image" src="https://github.com/user-attachments/assets/435ad751-5837-4b22-b981-59523b24f7aa" />